sdks/java/core/src/test/java/org/apache/beam/sdk/util/GcsUtilTest.java

```diff
@@ -321,28 +321,72 @@ public void testAccessDeniedObjectThrowsIOException() throws IOException {
-        new StorageObject().setSize(BigInteger.valueOf(1000)));
-
-    assertEquals(1000, gcsUtil.fileSize(GcsPath.fromComponents("testbucket", "testobject")));
+    String singleFile = "--batch_foobarbaz\n"
+        + "Content-Type: application/http\n"
+        + "\n"
+        + "HTTP/1.1 200 OK\n"
+        + "Content-Length: 708\n"
+        + "\n"
+        + "{\n"
+        + " \"kind\": \"storage#object\",\n"
+        + " \"id\": \"apache-beam-samples/apache/LICENSE/1470763581808000\",\n"
+        + " \"selfLink\": \"https://www.googleapis.com/storage/v1/b/apache-beam-samples/o/apache%2FLICENSE\",\n"
+        + " \"name\": \"apache/LICENSE\",\n"
+        + " \"bucket\": \"apache-beam-samples\",\n"
+        + " \"generation\": \"1470763581808000\",\n"
+        + " \"metageneration\": \"2\",\n"
+        + " \"contentType\": \"application/txt\",\n"
+        + " \"timeCreated\": \"2016-08-09T17:26:21.801Z\",\n"
+        + " \"updated\": \"2016-08-09T17:26:41.935Z\",\n"
+        + " \"storageClass\": \"STANDARD\",\n"
+        + " \"size\": \"11358\",\n"
+        + " \"md5Hash\": \"O4Pvljh/FGVfyFTdw8a9Vw==\",\n"
+        + " \"mediaLink\": \"https://www.googleapis.com/download/storage/v1/b/apache-beam-samples/o/apache%2FLICENSE?generation=1470763581808000&alt=media\",\n"
+        + " \"crc32c\": \"4W4HuQ==\",\n"
+        + " \"etag\": \"CIC73qbttM4CEAI=\"\n"
+        + "}\n"
+        + "\n";
+
+    int fileNum = 10;
+    List<GcsPath> paths = Lists.newArrayList();
+    StringBuilder content = new StringBuilder();
+    for (int i = 0; i < fileNum; ++i) {
+      content.append(singleFile);
+      paths.add(GcsPath.fromComponents("testbucket", "testobject" + i));
+    }
+    content.append("--batch_foobarbaz--\n");
+    List<Long> actuals = executeGetSizeBytes(paths, content.toString());
+    assertEquals(10, actuals.size());
+    assertEquals(11358, actuals.get(0).longValue());
   }
 
   @Test
   public void testGetSizeBytesWhenFileNotFound() throws Exception {
-    MockLowLevelHttpResponse notFoundResponse = new MockLowLevelHttpResponse();
-    notFoundResponse.setContent("");
-    notFoundResponse.setStatusCode(HttpStatusCodes.STATUS_CODE_NOT_FOUND);
+    String content = "Content-Type: application/http\n"
+        + "\n"
+        + "HTTP/1.1 404 Not Found\n"
+        + "Content-Length: 105\n"
+        + "\n"
+        + "{\n"
+        + " \"error\": {\n"
+        + "  \"code\": 404\n"
+        + " }\n"
+        + "}\n"
+        + "\n"
+        + "--batch_CnzAnRit7rQ_AAax2S5XfdM--\n";
+    thrown.expect(FileNotFoundException.class);
+    executeGetSizeBytes(
+        ImmutableList.of(GcsPath.fromComponents("testbucket", "testobject")),
+        content);
+  }
+
+  private List<Long> executeGetSizeBytes(
+      List<GcsPath> paths,
+      String content) throws Exception {
+    MockLowLevelHttpResponse notFoundResponse = new MockLowLevelHttpResponse()
+        .setContentType("multipart/mixed; boundary=batch_foobarbaz")
+        .setContent(content)
```

Comment:
Here, where you mock the low level response, I think you should either omit or fill in junk for pieces that `GcsUtil` should not depend on. But you shouldn't mock the low level response unless our code relies on it. Instead, mock the `storageClient`.